### PR TITLE
Use modern enums in compiler

### PR DESCRIPTION
**Pull request by cooldome**
Reduce number of constant strings, but reusing string representation of nim enums that compiler will generate anyway.
I managed to shave off 35 kb from the compiler binary with this trick.

I haven't noticed a slowdown in compiler despite I know about `reprEnum` string allocation overhead.
Possibly we need to improve codegen for `reprEnum` before making this change.

Also I think we need to get rid of `findStr` proc in wordrecg in favour of `parseEnum`. It is efficient now

### CODE REVIEW

**Comment by Clyybber** [comment]
I did similar things in https://github.com/nim-lang/Nim/pull/14777#issuecomment-648994776, see Araqs comment for why I reverted those changes in that PR.

**Comment by timotheecour** [comment]
That's now 3 people (i had also attempted a similar thing some time ago) that have tried to replace those `array[Enum, string]`. Unfortunately we're going in circles. See https://github.com/nim-lang/Nim/pull/14777#issuecomment-649050760

>>> I don't like the enumValue = "string repr" pattern

>> Can we then reconsider #14008 ?

> Yeah, that's a nice compromise. Bring it on please!

but then it got closed (along with its fusion alternative).

**Comment by Araq** [comment]
> That's now 3 people

And that's my threshold so I'm willing to accept this solution now.

**Comment by cooldome** [comment]
Looks good now, I will look into speeding up reprEnum in future PRs

File: compiler/ast.nim [hunk 1]
```diff
@@ -17,21 +17,17 @@ export int128
...
-    ccNoInline                # proc should not be inlined
-    ccFastCall                # fastcall (pass parameters in registers)
-    ccThisCall                # thiscall (parameters are pushed right-to-left)
-    ccClosure                 # proc has a closure
-    ccNoConvention            # needed for generating proper C procs sometimes
-
-const CallingConvToStr*: array[TCallingConvention, string] = ["nimcall", "stdcall",
-  "cdecl", "safecall", "syscall", "inline", "noinline", "fastcall", "thiscall",
-  "closure", "noconv"]
+    ccNimCall = "nimcall"           # nimcall, also the default
```
Comment (timotheecour) [inline compiler/ast.nim:20]: unfortunately this still leaves `CallingConvToStr: array[TCallingConvention, string] = ["N_NIMCALL",...` in compiler/ccgtypes.nim but without 
https://github.com/nim-lang/Nim/pull/14008/files#diff-6b7c28355957ab974fa38584bc810d1aac557bdcc7874e98d8ec005cc2200d72 that's the best one can do; still a good improvement.

**Review by timotheecour — APPROVED** [review]

**Pull request merged**